zstacklib/zstacklib/utils/gpu.py

```
@@ -539,3 +541,64 @@ def get_gpu_status_cmd(pci_device_address, iswindows=False):
+
+def has_nvidia_gpu():
+    r, _, _ = bash_roe("which nvidia-smi")
+    if r != 0:
+        return False
+    r, o, e = bash_roe("nvidia-smi -L")
+    return r == 0 and o and len(o.strip()) > 0
+
+
+def ensure_nvidia_persistenced_once(timeout=5):
+    # If already running, nothing to do
+    r, o, e = bash_roe("pgrep -x nvidia-persistenced || true")
```

Ruff (0.14.2)

`bash_roe` may be undefined, or defined from star imports

(F405)